### ISSUE TITLE
consolidate JSON saving form fields with a base class

### ISSUE THREAD

**Issue by NightJar** [silverstripe/silverstripe-ckan-registry#65]
As per point 1 in #57

@robbieaverill points out
>PHP FormField implementations: all three FormFields are essentially composite React component fields which read and write their value via serialised JSON. We could probably centralise some of the logic for this between the three classes.

@ScopeyNZ follows up
>Good idea. I almost feel like there could be something in core for this JsonComponentField or something. You could just extend it and define what component is used...

@NightJar continues
>A base "JSONField" or "JSONStoreField" or something - one can the use setComponent or define services for using different react components.

In summary:

Create a form field type that will save JSON blobs to the database. Then continue to replace `ResourceLocatorField`, `ResultConditionsField`, and `PresentedOptionsField` (in an upcoming PR) with this base field, setting the react component property to use the relevant view.

**Comment by robbieaverill** [comment]
I don't think it makes sense to do this until #25 is completed, since `ResultConditionsField` just stores its JSON value as a string which is entirely handled on the frontend - there's no logic there. We'll otherwise be refactoring for one use case (`ResourceLocatorField`) at this stage.

**Comment by robbieaverill** [comment]
Similarly to the other issue, do we still want to do this? I'm happy to close it

**Comment by ScopeyNZ** [comment]
I think it's not that important on this repo. It's a good idea for `silverstripe-admin` though.

**Comment by NightJar** [comment]
I think it's not necessary now, but I don't think it's worth closing either (unless as @ScopeyNZ says, it's moved to `silverstripe/admin` instead). It's certainly something that should be tracked.
